python/pysrc/kaskada/_timestream.py

```diff
@@ -1192,6 +1194,69 @@ def run_iter(
 
         raise AssertionError(f"Unhandled kind {kind}")
 
+    def plan(
+        self,
+        kind: Literal["initial_dfg", "final_dfg", "final_plan"] = "final_plan",
+        results: Optional[Union[kd.results.History, kd.results.Snapshot]] = None,
+        format: Literal["dot", "svg", "svg_str"] = "svg",
+        mode: Literal["once", "live"] = "once",
+    ) -> Union[str, "IPython.display.SVG"]:
+        """Return GraphViz representation of the DFG execution plan.
+
+        This is intended for understanding how a given Timestream query will
+        be executed. It does not guarantee that the same query will always
+        be executed the same way.
+
+        Args:
+            results: The results to produce. Defaults to `History()` producing all points.
+            kind: The kind of plan to produce.
+            format: Specify how the output should be returned.
+              If `dot`, the GraphViz `dot` format is returned as a string.
+              If `svg_str`, the GraphViz is rendered to an SVG string.
+              If `svg`, the GraphViz is rendered to an SVG image.
+            mode: The execution mode to use. Defaults to `'once'` to produce the results
+              from the currently available data. Use `'live'` to start a standing query
+              that continues to process new data until stopped.
+        Returns:
+            A GraphViz representation of the execution plan.
+            Specific representation depends on the `format` argument.
+
+        Raises:
+            ValueError if the `kind` is not recognized or the `format` is not supported.
+        """
+        expr = self
+        if not pa.types.is_struct(self.data_type):
+            # The execution engine requires a struct, so wrap this in a record.
+            expr = record({"result": self})
+
+        options = _ExecutionOptions.create(
+            results, row_limit=None, max_batch_size=None, mode=mode
+        )
+
+        dot = expr._ffi_expr.plan(kind, options)
+        if format == "dot":
+            return dot
+
+        svg_str = ""
+        try:
```

Comment:
This pattern of importing within a try block ensures that we don't require it for other files. It also lets us report an error saying "this package needs to be installed for this option".